src/main/scala/com/databricks/labs/overwatch/ApiCallV2.scala

```diff
@@ -619,6 +625,124 @@ class ApiCallV2(apiEnv: ApiEnv) extends SparkSessionWrapper {
     newJsonObject.toString
   }
 
+  /**
+   * Function to make parallel API calls. Currently this functions supports only SqlQueryHistory and ClusterEvents
+   * @param endpoint
+   * @param jsonInput
+   * @param config
+   * @return
+   */
+  def makeParallelApiCalls(endpoint: String, jsonInput: Map[String, String], config: Config): String = {
```

Comment:
I really think this belongs in a class and that we should refactor APICall, APICallV2, and APIMeta and move them into a new namespace (folder) like API. Inside of API perhaps we create something like `APIParHandler`. There are so many variables at the top which look like class vars and then we could break up the future section into several smaller functions to simplify support for this. Thoughts? This really isn't me telling you to do this, looking for discussion / thoughts / feedback on this idea. @sriram251-code, I would like your feedback too.